stonesoup/types/interval.py

```diff
@@ -0,0 +1,336 @@
+
+    def __missing__(self, key):
+        raise NotImplementedError('Cannot index a continuous interval')
+
+    def __iter__(self):
+        raise NotImplementedError('Cannot index a continuous interval. Use the self.element_iter()'
+                                  ' or self.interval_iter() method')
+
+    def element_iter(self, spacing):
+        if spacing > self.length:
+            raise ValueError('Spacing must be smaller than, or equal to the size of the interval')
+        element_list = []
+        value = self.left
+        while value <= self.right:
+            element_list.append(value)
+            value += spacing
+        return iter(element_list)
+
+    def interval_iter(self, interval_size):
+        if interval_size > self.length:
+            raise ValueError('Sub-interval size must be smaller than, or equal to the size of the'
+                             ' interval')
+        interval_list = []
+        current_interval = ClosedContinuousInterval(self.left, self.left + interval_size)
+        while current_interval.right <= self.right:
+            interval_list.append(current_interval)
+            current_interval = ClosedContinuousInterval(current_interval.right,
+                                                        current_interval.right + interval_size)
+        return iter(interval_list)
+
+    def __reversed__(self):
+        raise NotImplementedError('Cannot index a continuous interval. Use the'
+                                  'self.reversed_element_iter() or self.reversed_interval_iter()'
+                                  'method')
+
+    def reversed_element_iter(self, spacing):
+        if spacing > self.length:
+            raise ValueError('Spacing must be smaller than, or equal to the size of the interval')
+        element_list = []
+        value = self.right
+        while value >= self.left:
+            element_list.append(value)
+            value -= spacing
+        return iter(element_list)
+
+    def reversed_interval_iter(self, interval_size):
+        if interval_size > self.length:
+            raise ValueError('Sub-interval size must be smaller than, or equal to the size of the'
+                             ' interval')
+        interval_list = []
+        current_interval = ClosedContinuousInterval(self.right - interval_size, self.right)
+        while current_interval.left >= self.left:
+            interval_list.append(current_interval)
+            current_interval = ClosedContinuousInterval(current_interval.left - interval_size,
+                                                        current_interval.left)
+        return iter(interval_list)
+
+    def __contains__(self, item):
+        if isinstance(item, Real):
+            return self.left <= item <= self.right
+        else:
+            return False
+
+    def __str__(self):
+        return '[' + str(self.left) + ', ' + str(self.right) + ']'
+
+    def __add__(self, other):
```

Comment:
Wonder if this should be `union`/`|`/`__or__` like set type instead of `+`/`__add__`?

Could also include `intersection`/`&`/`__and__`?

Reply:
Yes, that makes more sense.